Mathlib/CategoryTheory/Limits/Shapes/SplitEqualizer.lean

```diff
@@ -0,0 +1,163 @@
+  /-- A map in the opposite direction to `f` and `g` -/
+  rightRetraction : Y ⟶ X
+  /-- Composition of `ι` with `f` and with `g` agree -/
+  condition : ι ≫ f = ι ≫ g := by aesop_cat
+  /-- `leftRetraction` splits `ι` -/
+  ι_leftRetraction : ι ≫ leftRetraction = 𝟙 W := by aesop_cat
+  /-- `rightRetraction` splits `g` -/
+  bottom_rightRetraction : g ≫ rightRetraction = 𝟙 X := by aesop_cat
+  /-- `f` composed with `rightRetraction` is `leftRetraction` composed with `ι` -/
+  top_rightRetraction : f ≫ rightRetraction = leftRetraction ≫ ι := by aesop_cat
+
+instance {X : C} : Inhabited (IsSplitEqualizer (𝟙 X) (𝟙 X) (𝟙 X)) where
+  default := { leftRetraction := 𝟙 X, rightRetraction := 𝟙 X }
+
+open IsSplitEqualizer
+
+attribute [reassoc] condition
+
+attribute [reassoc (attr := simp)] ι_leftRetraction bottom_rightRetraction top_rightRetraction
+
+variable {f g}
+
+/-- Split equalizers are absolute: they are preserved by any functor. -/
+@[simps]
+def IsSplitEqualizer.map {W : C} {ι : W ⟶ X} (q : IsSplitEqualizer f g ι) (F : C ⥤ D) :
+    IsSplitEqualizer (F.map f) (F.map g) (F.map ι) where
+  leftRetraction := F.map q.leftRetraction
+  rightRetraction := F.map q.rightRetraction
+  condition := by rw [← F.map_comp, q.condition, F.map_comp]
+  ι_leftRetraction := by rw [← F.map_comp, q.ι_leftRetraction, F.map_id]
+  bottom_rightRetraction := by rw [← F.map_comp, q.bottom_rightRetraction, F.map_id]
+  top_rightRetraction := by rw [← F.map_comp, q.top_rightRetraction, F.map_comp]
+
+section
+
+open Limits
+
+/-- A split equalizer clearly induces a fork. -/
+@[simps! pt]
+def IsSplitEqualizer.asFork {W : C} {h : W ⟶ X} (t : IsSplitEqualizer f g h) :
+    Fork f g := Fork.ofι h t.condition
+
+@[simp]
+theorem IsSplitEqualizer.asFork_ι {W : C} {h : W ⟶ X} (t : IsSplitEqualizer f g h) :
+    t.asFork.ι = h := rfl
+
+/--
+The fork induced by a split equalizer is an equalizer, justifying the name. In some cases it
+is more convenient to show a given fork is an equalizer by showing it is split.
+-/
+def IsSplitEqualizer.isEqualizer {W : C} {h : W ⟶ X} (t : IsSplitEqualizer f g h) :
+    IsLimit t.asFork :=
+  Fork.IsLimit.mk' _ fun s =>
+    ⟨ s.ι ≫ t.leftRetraction,
+      by simp [- top_rightRetraction, ← t.top_rightRetraction, s.condition_assoc],
+      fun hm => by simp [← hm] ⟩
+
+end
+variable (f g)
+
+/--
+The pair `f,g` is a cosplit pair if there is an `h : W ⟶ X` so that `f, g, h` forms a split
+equalizer in `C`.
+-/
+class HasSplitEqualizer : Prop where
+  /-- There is some split equalizer -/
+  splittable : ∃ (W : C) (h : W ⟶ X), Nonempty (IsSplitEqualizer f g h)
+
+/--
+The pair `f,g` is a `G`-cosplit pair if there is an `h : W ⟶ G X` so that `G f, G g, h` forms a
+split equalizer in `D`.
+-/
+abbrev Functor.IsCosplitPair : Prop :=
+  HasSplitEqualizer (G.map f) (G.map g)
+
+/-- Get the equalizer object from the typeclass `IsCosplitPair`. -/
+noncomputable def HasSplitEqualizer.equalizerOfCosplit [HasSplitEqualizer f g] : C :=
```

Comment:
This def should be called `HasSplitEqualizer.equalizer` or something like that, as it has nothing to do with `CosplitPair`. I guess it's adapted from the other file, in which case that should be changed too.

Reply:
Ok, I changed this name to `HasSplitEqualizer.equalizerOfSplit`, and didn't touch it in the other file. The `Split` in the name in the other file doesn't refer to `Functor.IsSplitPair`, hence we shouldn't use `Cosplit here.